lib/module.ts

```diff
@@ -59,15 +26,18 @@ export class Module<
     flattenedModule.__metadata = this.getMetadata()
 
     // state
-    const state = reactive(this.options.stateInit?.() ?? ({} as S))
+    const initialState = this.getInitialState()
+    const state = reactive(initialState ?? ({} as S))
+
+    // map state properties to Ref<T>
     for (const key in state) {
       // TODO: find the correct typing for this
       flattenedModule[key] = readonly(toRef(state, key)) as IFlattenedModule<
         S,
         G,
         M,
         SM
-      >[Extract<keyof UnwrapNestedRefs<S>, string>]
+      >['string']
```

Comment:
I couldn't figure out the best way to write this, I'm not sure I fully understand the types yet. I did notice that `Extract<keyof UnwrapNestedRefs<S>, string>` is equivalent to `'string'`, so I made that change.

As far as I can tell, this whole thing is basically equivalent to `readonly(toRef(state, key)) as any`, but I wasn't confident that it's the right solution.